docs/getting-started/istio/index.md

```diff
@@ -33,6 +33,8 @@ spec:
           virtualService:
             # Reference to a VirtualService which the controller updates with canary weights
             name: rollouts-demo-vsvc
+            # Optional if there are only two destinations in your routes or if you want to split 100% traffic between stable and canary services. If specified, this will be used as an upper bound for traffic between canary + stable services.
+            maxWeight: 80
```

Comment:
This PR adds maxWeight prop at the level of the virtualService, therefore it will affect all the routes ArgoRollouts manage - which are specified in strategy.canary.trafficRouting.istio.virtualService.routes

While in our use case we'd want only specific routes to have less then 100 weight managed by ArgoRollouts.
Our Virtual Service has such routes
- a route for the traffic originating in Mesh. Here we want e.g. 80% to be split between stable and canary in Mesh, and 20% to go to the legacy system.
- a route for the traffic coming to Mesh Ingress Gateway from the Legacy system. Here we want 100% to be split between stable and canary in Mesh. (we don't want to bounce the traffic back to the legacy)

What do you think? 

(maybe we can move the maxWeight to the level of the strategy.canary.trafficRouting.istio.virtualService.routes. Or remove maxWeight  and configure ArgoRollouts to manage only the first 2 destinations in founds in a route...)

Reply:
There is no straightforward way to do it at the routes level. Currently, "routes" is nothing but an array of string which takes the HTTP route names. For us to do it at the route-level, we need a struct/object similar to "tlsRoutes" where we can add an optional "maxWeight" prop.

One proposal is to add "httpRoute" which will be an array of object similar to "tlsRoutes" and gradually deprecate the use of "routes". I proposed this to @jessesuen as well.

@jessesuen @huikang Any suggestions on what would be the best thing to do here?

Reply:
Hi @agrawroh, I am interested in coming up with a solution for this as well. I would like to reopen this PR or something similar to support this feature. 

@jessesuen @huikang are there any suggestions for moving forward ?

Reply:
@dnguy078 Sure, that'd be great. Please feel free to take it over or let me know how I can help.

Reply:
Thank you, @agrawroh I am trying to solve an issue related to istio subset level in another PR as well. https://github.com/argoproj/argo-rollouts/pull/3126 I was wondering if you could take a quick look :)

Reply:
@dnguy078 any update? This is a pretty big blocker for argo-rollouts adoption on our end.